src/Symfony/Component/DependencyInjection/Attribute/Target.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * An attribute to tell how a dependency is used and hint named autowiring aliases.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class Target
+{
+    /**
+     * @var string
+     */
+    public $name;
+
+    public function __construct(string $name)
+    {
+        $this->name = lcfirst(str_replace(' ', '', ucwords(preg_replace('/[^a-zA-Z0-9\x7f-\xff]++/', ' ', $name))));
+    }
+
+    public static function getName(\ReflectionParameter $parameter): string
```

Comment:
I don't feel like this method should be on the attribute class. And the fact that this class has a property `name` and this method is named like it's a getter for that property adds unnecessary confusion.

This method looks like some helper logic that I need to build compiler passes, so maybe we can move this into a trait that the relevant passes can include.

Reply:
I'm not sure I understand your concern. To me, this method *should* be on the attribute class, because SRP. A trait would basically say that every consumer of the attribute needs to reimplement the same logic, from a design pov.

The method is a helper to extract and interpret the attribute. And to me, attributes are legit classes to convey this knowledge.

Reply:
> A trait would basically say that every consumer of the attribute needs to reimplement the same logic, from a design pov.

No, it wouldn't. A trait encapsulates shared logic. If you don't need it, you don't use it.

> The method is a helper to extract and interpret the attribute. And to me, attributes are legit classes to convey this knowledge.

Fine, let's keep it here, but may I suggest to rename the method to e.g. `extractName()` or `parseName()`? I think it's the misleading usage of the getter naming convention that bothers me most.

Reply:
Works for me, thanks for the suggestion. Now updated to `parseName()`.